tests/console/java.pm

```diff
@@ -21,56 +21,35 @@ use base "consoletest";
 
 sub run {
-    select_console 'root-console';
-
+    my ($self) = @_;
+    $self->select_serial_terminal;
     # Make sure that PackageKit is not running
     pkcon_quit;
+    # Supported Java versions for sle15sp2
+    # https://www.suse.com/releasenotes/x86_64/SUSE-SLES/15-SP2/#development-java-versions
+    # java-11-openjdk                   -> Basesystem
+    # java-10-openjdk & java-1_8_0-ibm  -> Legacy
+    my $cmd = 'install --auto-agree-with-licenses ';
 
-    if (check_var("DISTRI", "sle")) {
-        if (zypper_call('in --auto-agree-with-licenses java-*', timeout => 500, exitcode => [0, 4])) {
-            # install only java-11-openjdk* & java-*-ibm*
-            zypper_call('in --auto-agree-with-licenses java-11-openjdk* java-*-ibm*');
-        }
-    }
-
-    if (check_var("DISTRI", "opensuse")) {
-        # Capture the return code value of the following scenarios
-        my $bootstrap_pkg_rt       = zypper_call("se java-*bootstrap",                             exitcode => [0, 104]);
-        my $bootstrap_conflicts_rt = zypper_call("in --auto-agree-with-licenses --dry-run java-*", exitcode => [0, 4]);
-
-        # logs / debugging purposes
-        diag "checking variable: bootstrap_pkg_rt = $bootstrap_pkg_rt";
-        diag "checking variable: bootstrap_conflicts_rt = $bootstrap_conflicts_rt";
+    $cmd .= (is_sle('15+') || is_leap) ? 'java-11-openjdk* java-1_*' : 'java-*';
+    zypper_call($cmd);
 
-        if ($bootstrap_pkg_rt == 0) {
-            diag "There are java bootstrap packages available to be installed";
-            print "There are java bootstrap packages available to be installed\n";
-            if ($bootstrap_conflicts_rt == 0) {
-                diag "There is no conflict installing the java bootstrap packages";
-                print "There is no conflict installing the java bootstrap packages\n";
-                zypper_call("in java-*");
-            }
-            else {
-                diag "There are conflicts with the installation of java bootstrap packages";
-                print "There are conflicts with the installation of java bootstrap packages\n";
-                record_soft_failure 'boo#1019090';
-                # Workaround: install java-* except from the problematic bootstrap packages
-                zypper_call "in `(zypper se java-* | grep -v bootstrap | grep -v 'i ' | awk '{print \$2}' | sed -n -E -e '/java/,\$ p')`";
-            }
-        }
-        else {
-            diag "There are no java bootstrap packages";
-            print "There are no java bootstrap packages\n";
-            zypper_call("in java-*");
-        }
+    if (script_run 'rpm -q wget') {
+        zypper_call 'in wget';
     }
-
-    zypper_call 'in wget';
     assert_script_run 'wget --quiet ' . data_url('console/test_java.sh');
     assert_script_run 'chmod +x test_java.sh';
-    assert_script_run './test_java.sh';
+    assert_script_run qq{./test_java.sh | tee -a /dev/$serialdev};
```

Comment:
Is `pipefail` set in this shell? Otherwise this won't detect errors

Reply:
No, it is handled in `script_output` to mine knowledge. Good that you have noticed it, thanks. 
Regarding sle '15+' and java testing. QAM enables all repositories in their tests, therefore their test suites pull java from legacy as well. In JeOS sle15sp2/sle15sp2 we do not really test the legacy module. 
Since the test case purpose is to install multiple java versions, I am about to enable the legacy module in `jeos-base+sdk` test suite or add legacy module registration/deregistration directly in the test case.